Patch-Source: https://github.com/matrix-org/pantalaimon/commit/3968c69aa846889970df1372ba9aa54c1c5e4290.patch
--
From 3968c69aa846889970df1372ba9aa54c1c5e4290 Mon Sep 17 00:00:00 2001
From: Igor Artemenko <igor@codemonium.com>
Date: Wed, 5 Apr 2023 16:17:04 +0000
Subject: [PATCH] Set Pantalaimon presence to offline

Before this change, Pantalaimon users would always appear online because
that is the default state when the /sync endpoint's set_presence
parameter is not set. By explicitly setting the parameter to "offline",
only the user-facing client (which executes its own /sync request)
affects the presence state.
---
 pantalaimon/client.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pantalaimon/client.py b/pantalaimon/client.py
index be32859..ececee3 100644
--- a/pantalaimon/client.py
+++ b/pantalaimon/client.py
@@ -554,6 +554,7 @@ def start_loop(self, loop_sleep_time=100):
                 full_state=True,
                 since=next_batch,
                 loop_sleep_time=loop_sleep_time,
+                set_presence="offline",
             )
         )
         self.task = task
